assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js

```diff
@@ -130,137 +140,144 @@ function SizePositionPanel({
-      )}
-      {/** X/Y */}
-      <Row expand>
-        <BoxedNumeric
-          suffix={_x('X', 'Position on X axis', 'web-stories')}
-          value={x}
-          min={minMaxXY.minX}
-          max={minMaxXY.maxX}
-          onChange={(value) => pushUpdate({ x: value })}
-          aria-label={__('X position', 'web-stories')}
-          canBeNegative
-        />
-        <Spacer />
-        <BoxedNumeric
-          suffix={_x('Y', 'Position on Y axis', 'web-stories')}
-          value={y}
-          min={minMaxXY.minY}
-          max={minMaxXY.maxY}
-          onChange={(value) => pushUpdate({ y: value })}
-          aria-label={__('Y position', 'web-stories')}
-          canBeNegative
-        />
-      </Row>
-      {/** Width/height & lock ratio */}
-      <Row expand>
-        <BoxedNumeric
-          suffix={_x('W', 'The Width dimension', 'web-stories')}
-          value={width}
-          min={MIN_MAX.WIDTH.MIN}
-          max={MIN_MAX.WIDTH.MAX}
-          onChange={(value) => {
-            const newWidth = value;
-            let newHeight = height;
-            if (lockAspectRatio) {
-              if (newWidth === '') {
-                newHeight = '';
-              } else if (isNum(newWidth / origRatio)) {
-                newHeight = dataPixels(newWidth / origRatio);
+      <Grid>
+        <Area area="x">
+          <NumericInput
+            suffix={_x('X', 'Position on X axis', 'web-stories')}
+            value={x}
+            min={minMaxXY.minX}
+            max={minMaxXY.maxX}
+            onChange={(evt) => {
+              const value = Number(evt.target.value);
+              pushUpdate({ x: value }, true);
+            }}
+            aria-label={__('X position', 'web-stories')}
+            canBeNegative
+            {...getMixedValueProps(x)}
+          />
+        </Area>
+        <Area area="y">
+          <NumericInput
+            suffix={_x('Y', 'Position on Y axis', 'web-stories')}
+            value={y}
+            min={minMaxXY.minY}
+            max={minMaxXY.maxY}
+            onChange={(evt) => {
+              const value = Number(evt.target.value);
+              pushUpdate({ y: value }, true);
+            }}
+            aria-label={__('Y position', 'web-stories')}
+            canBeNegative
+            {...getMixedValueProps(y)}
+          />
+        </Area>
+        {/** Width/height & lock ratio */}
+        <Area area="w">
+          <NumericInput
+            suffix={_x('W', 'The Width dimension', 'web-stories')}
+            value={width}
+            min={MIN_MAX.WIDTH.MIN}
+            max={MIN_MAX.WIDTH.MAX}
+            onChange={(evt) => {
+              const newWidth = Number(evt.target.value);
+              let newHeight = height;
+              if (lockAspectRatio) {
+                if (newWidth === '') {
+                  newHeight = '';
+                } else if (isNum(newWidth / origRatio)) {
+                  newHeight = dataPixels(newWidth / origRatio);
+                }
```

Comment:
nit: for readability we could move this out into its own event handler function? If we wrap it in a `useCallback` we also don't have to worry about this function being re-created every render. It's a little large for an inline function imo, but it's no big deal either way!